httputil/request.go

```diff
@@ -172,14 +180,19 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
-	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	for _, headerName := range []string{WwwAuthenticateHeader, LfsAuthenticateHeader} {
```

Comment:
I agree. I was thinking about this, but was not sure what is the right way to do this in GO lang. Thanks for teaching me best practices.